src/develop/payment.md

```diff
@@ -0,0 +1,272 @@
+---
+sidebar_position: 12
+description: Integrating Evercloud GraphQL API with Typescript
+---
+
+# GraphQL code generation
+
+This is a guide to help you quickly integrate Everscale into your Typescript project using Evercloud GraphQL API for making queries to the blockchain, and [GraphQL Code Generator](https://the-guild.dev/graphql/codegen) for automatically generating type definitions from the API schema.
+
+## Evercloud GraphQL
+
+[Evercloud](https://www.evercloud.dev/) makes it easy to set up and manage a GraphQL endpoint for your application, providing you with secure access to the Everscale blockchain.
+
+Follow [this guide](https://docs.evercloud.dev/products/evercloud/get-started) to set up a project on Evercloud. Make sure to note down the project ID in the security tab, as well as the project secret, as you will need to configure these in the code.
+
+Once you have the necessary credentials, you're ready to start making GraphQL queries to the Everscale network in your TypeScript project. To cater to various development preferences, we'll provide two versions for each example: one using the Everscale SDK (follow [this guide](https://docs.everos.dev/ever-sdk/guides/installation/add_sdk_to_your_app) for setup instructions) and another using [Axios](https://axios-http.com/), a popular HTTP client for JavaScript.
+
+## Setting up code generation
+
+First, add both the `graphql` and `@graphql-codegen/cli` packages to your project's dependencies:
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+<Tabs>
+  <TabItem value="npm" label="npm" default>
+
+```shell
+npm install graphql
+npm install -D typescript
+npm install -D @graphql-codegen/cli
+npm install -D @graphql-codegen/typescript
+```
+  </TabItem>
+  <TabItem value="yarn" label="yarn">
+
+```shell
+yarn add graphql
+yarn add -D typescript
+yarn add -D @graphql-codegen/cli
+yarn add -D @graphql-codegen/typescript
+```
+  </TabItem>
+</Tabs>
+
+Next, create a file named `codegen.ts` in your project's root folder containing the following (adjust as needed):
+
+```typescript
+import type {CodegenConfig} from '@graphql-codegen/cli'
+
+// configure your credentials here
+const PROJECT_ID = ''
+const PROJECT_SECRET = ''
+
+const schemaUrl = `https://mainnet.evercloud.dev/${PROJECT_ID}/graphql`
+
+const config: CodegenConfig = {
+  overwrite: true,
+  schema: [
+    {
+      [schemaUrl]: {
+        headers: {
+          Authorization: `Basic ${PROJECT_SECRET}`,
```

Comment:
PROJECT_SECRET_IN_BASE64

Reply:
I've changed the name of the variable, as well as added a clarification in the instructions.